src/main/java/com/google/cloud/spanner/jdbc/JdbcParameterStore.java

```diff
@@ -600,6 +671,25 @@ private Builder setParamWithKnownType(ValueBinder<Builder> binder, Object value,
     return null;
   }
 
+  private String getStringFromInputStream(InputStream inputStream) throws SQLException {
+    InputStreamReader reader = new InputStreamReader(inputStream, StandardCharsets.US_ASCII);
```

Comment:
Why do we use `ASCII` instead of `UTF-8` here?

Reply:
cc @zoercai

Reply:
Thanks @thiagotnunes , yep String values passed to the backend must be UTF-8 encoded

Reply:
This method is used for this (required JDBC interface) method: https://github.com/googleapis/java-spanner-jdbc/blob/4ed17f0061077505d772c89589a0965f3bbc4a18/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java#L361

There's also a similar Unicode version which is called [setCharacterStream(..)](https://github.com/googleapis/java-spanner-jdbc/blob/4ed17f0061077505d772c89589a0965f3bbc4a18/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java#L373), which expects the input to be in Unicode format.

The data is __read__ as ASCII, but encoded as UTF-8 when it is sent to Cloud Spanner.

Reply:
Ah ok, this method is only used in the `setAsciiStream` then? I guess that should be fine if that is required by JDBC.

Reply:
> Ah ok, this method is only used in the `setAsciiStream` then? I guess that should be fine if that is required by JDBC.

Yes, this is only used by the `setAsciiStream` method. JDBC unfortunately contains a number of old method definitions that still need to be supported.